cmd/root.go

```diff
@@ -73,37 +65,10 @@ func NewRootCmd() *cobra.Command {
     \__\/\__\/ \_____\/   \__\/    \_\/ \_\/ \_____\/           \_____\/ \_____\/\________\/
 
 Welcome to the Astro CLI, the modern command line interface for data orchestration. You can use it for Astro, Astronomer Software, or Local Development.`,
-		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			// Check for latest version
-			if config.CFG.UpgradeMessage.GetBool() {
-				// create github client with 3 second timeout, setting an aggressive timeout since its not mandatory to get a response in each command execution
-				githubClient := github.NewClient(&http.Client{Timeout: 3 * time.Second})
-				// compare current version to latest
-				err = version.CompareVersions(githubClient, "astronomer", "astro-cli")
-				if err != nil {
-					softwareCmd.InitDebugLogs = append(softwareCmd.InitDebugLogs, "Error comparing CLI versions: "+err.Error())
-				}
-			}
-			if isCloudCtx {
-				err = cloudCmd.Setup(cmd, platformCoreClient, astroCoreClient)
-				if err != nil {
-					if strings.Contains(err.Error(), "token is invalid or malformed") {
-						return errors.New("API Token is invalid or malformed") //nolint
-					}
-					if strings.Contains(err.Error(), "the API token given has expired") {
-						return errors.New("API Token is expired") //nolint
-					}
-					softwareCmd.InitDebugLogs = append(softwareCmd.InitDebugLogs, "Error during cmd setup: "+err.Error())
-				}
-			}
-			// common PersistentPreRunE component between software & cloud
-			// setting up log verbosity and dumping debug logs collected during CLI-initialization
-			if err := softwareCmd.SetUpLogs(os.Stdout, verboseLevel); err != nil {
-				return err
-			}
-			softwareCmd.PrintDebugLogs()
-			return nil
-		},
+		PersistentPreRunE: utils.ChainRunEs(
+			SetupLoggingPersistentPreRunE,
+			CreateRootPersistentPreRunE(astroCoreClient, platformCoreClient),
+		),
```

Comment:
We've broken apart the core of the function and the piece that sets up the logger into 2 separate hook functions, which we chain together here. Functionally, nothing changes, but we've built a pattern we can expand on in a future PR.